policy-controller/core/src/outbound.rs

```diff
@@ -22,20 +31,40 @@ pub type OutboundPolicyStream = Pin<Box<dyn Stream<Item = OutboundPolicy> + Send
 
 pub type HttpRoute = OutboundRoute<HttpRouteMatch, HttpRetryCondition>;
 pub type GrpcRoute = OutboundRoute<GrpcRouteMatch, GrpcRetryCondition>;
+
 pub type RouteSet<T> = HashMap<GroupKindNamespaceName, T>;
 
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub enum TrafficPolicy {
+    Allow,
+    Deny,
+}
+
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub enum Kind {
+    EgressNetwork {
+        original_dst: SocketAddr,
+        traffic_policy: TrafficPolicy,
```

Comment:
This actually leads to a more difficult question about EgressNetwork mutability and ip address.

For services, a service's IP address will never change for the life of that services which means we can look up the service by IP once at the beginning of the discovery stream and never again.

For EgressNetworks, on the other hand, they encompass a range of IP addresses and that range can change if the EgressNetwork resource is edited.  What happens when a discovery stream is started on an IP address that is in an EgressNetwork, but then the EgressNetwork is edited to no longer include that IP in it's network range?  Or vice versa, what if an EgressNetwork is edited to include an IP address that was used for an already started discovery stream?

Reply:
It really seems like we need to take a different approach for EgressNetworks compared to the approach we have for services in the outbound index.  Perhaps we can take advantage of the fact that cluster networks are immutable. When a discovery lookup comes in, we can look at if it's in the cluster network and send it to the service index or if it's out of the cluster network, then we can send it to the egress index.  

The egress index will likely need to have watches indexed by ip address so that we can update the appropriate watches whenever an EgressNetwork resource changes and may encompass watches that it did not before or stop encompassing watches that it used to.

Reply:
Good points.. Will think about it a bit more.